opencti-platform/opencti-graphql/src/migrations/1678399004325-roles-under-groups.js

```diff
@@ -1,23 +1,23 @@
 import { executionContext, SYSTEM_USER } from '../utils/access';
 import { logApp } from '../config/conf';
-import { findAll, findRoles, userAddRelation, userIdDeleteRelation } from '../domain/user';
-import { addGroup, groupAddRelation } from '../domain/group';
+import { findAll as findUsers, findRoles, userAddRelation, userIdDeleteRelation } from '../domain/user';
+import { addGroup, findAll as findGroups, groupAddRelation } from '../domain/group';
 
 export const up = async (next) => {
   const context = executionContext('migration', SYSTEM_USER);
   const start = new Date().getTime();
   logApp.info('[MIGRATION] Refacto roles under groups');
-  const users = await findAll(context, context.user);
+  const [users, roles, groups] = await Promise.all([findUsers(context, context.user), findRoles(context, context.user), findGroups(context, context.user)]);
   const usersNodes = users.edges.map((user) => user.node);
-  const roles = await findRoles(context, context.user);
   const rolesNodes = roles.edges.map((role) => role.node);
+  const groupsNames = groups.edges.map((group) => group.node).map((group) => group.name);
   let roleGroupAssociations = {};
   // eslint-disable-next-line no-restricted-syntax
```

Comment:
Can you rewrite this correctly at the same time ?

Reply:
Done